### DISCUSSION TITLE
Can't use transition; 'Object reference not set to an instance of an object.'

### DISCUSSION THREAD

**Discussion by JamDoggie** [AvaloniaUI/Avalonia#5899] [Q&A] [answered]
So basically I can't use transitions without the program instantly crashing with the very undescriptive error in the title, pointing at the DoubleTransition. Am I doing something wrong?

Steps to reproduce:
- Create a blank Avalonia project on latest version, with Visual Studio 2019
- Replace the "Hello Avalonia" text with this:

```
<Button Content="Test" HorizontalAlignment="Center">
    <Button.Transitions>
      <DoubleTransition Property="Opacity" Duration="0:0:0.2"/>
    </Button.Transitions>
  </Button>
```
  

  Any thoughts?

**Comment by maxkatz6** [comment, marked as answer]
```xaml
<Button Content="Test" HorizontalAlignment="Center">
    <Button.Transitions>
      <Transitions>
         <DoubleTransition Property="Opacity" Duration="0:0:0.2"/>
      </Transitions>
    </Button.Transitions>
  </Button>
```

**Reply by maxkatz6** [reply]
Button.Transitions is null by default, you need to set new object here to get it work

**Reply by JamDoggie** [reply]
Aha! I think it's time for someone to update the docs then https://avaloniaui.net/docs/animations/transitions

**Reply by JamDoggie** [reply]
..unless I was reading them wrong

**Reply by maxkatz6** [reply]
Thanks, updated new docs page
https://docs.avaloniaui.net/docs/animations/transitions

**Reply by JamDoggie** [reply]
Woah! I'm not sure that comes up on google, that's a much nicer looking set of docs!

**Reply by danwalmsley** [reply]
@kekekeks we need to fix that in xaml compiler.

**Reply by maxkatz6** [reply]
@danwalmsley I think there is bigger related issue. It might not be possible for compiler to understand, what exactly it should do - add new transitions or create new array of them.
https://github.com/AvaloniaUI/Avalonia/issues/4464
